### PR TITLE
PIM-4717: How to change the rules when importing reference files ?

### PR DESCRIPTION
**Pull request by sumbobyboys**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,8 +24,7 @@ Mac OS
 
 Download composer `curl -s https://getcomposer.org/installer | php` and run `php composer.phar install`.
 
-Build the documention
----------------------
+## Build the documention
```
Comment (phaseinducer) [inline README.md]: *documentation

File: cookbook/product_asset/change_upload_validation_rule.rst [hunk 2]
```diff
@@ -0,0 +1,91 @@
+How to change the validation rule to match a reference file to an asset
+=======================================================================
+
+**This cookbook is about a feature only provided in the Enterprise Edition.**
+
+To add files to assets in Akeneo we use a simple mapping strategy to match a filename to a product asset. As an integrator you can customize this rule to fits your own needs.
```
Comment (phaseinducer) [inline cookbook/product_asset/change_upload_validation_rule.rst]: *to fit

File: cookbook/product_asset/change_upload_validation_rule.rst [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+Let say that we only have non localizable assets and want to add a suffix to every file we import and remove the 3 first letters of their names.
```
Comment (phaseinducer) [inline cookbook/product_asset/change_upload_validation_rule.rst]: *Let's say

File: cookbook/product_asset/change_upload_validation_rule.rst [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+    use PimEnterprise\Component\ProductAsset\Upload\ParsedFilename as BaseParsedFilename;
+    use Pim\Bundle\CatalogBundle\Model\LocaleInterface;
+
+    /**
+     * @see FilenameParserInterface
+     */
+    class ParsedFilename extends BaseParsedFilename
+    {
+
```
Comment (phaseinducer) [inline cookbook/product_asset/change_upload_validation_rule.rst]: extra blank line

File: cookbook/product_asset/change_upload_validation_rule.rst [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+            $this->parseRawFilename($this->rawFilename);
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function parseRawFilename($rawFilename)
+        {
+            if (preg_match('/^(?P<name>[0-9A-Za-z-_]+)\.(?P<extension>[0-9A-Za-z]+)$/m', $rawFilename, $matches)) {
```
Comment (phaseinducer) [inline cookbook/product_asset/change_upload_validation_rule.rst]: I don't understand why we use the "m" option here.

File: cookbook/product_asset/change_upload_validation_rule.rst [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+                $customName = substr($matches['name'], 3) . '_custom_suffix';
+
+                $this->assetCode  = strlen($matches['name']) > 0 ? $customName : null;
+                $this->localeCode = 'en_US';
+                $this->extension  = $matches['extension'];
+            }
+        }
+    }
+
+You can now imagine impement your own logic and customize this rule as you want.
```
Comment (jmleroux) [inline cookbook/product_asset/change_upload_validation_rule.rst]: implement

Comment (phaseinducer) [inline cookbook/product_asset/change_upload_validation_rule.rst]: "how to implement" ?

**Comment by solivier** [comment]
GTM once comments fixed

**Comment by jmleroux** [comment]
GTM

**Pull request merged**